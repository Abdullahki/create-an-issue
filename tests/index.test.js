const path = require('path')
const IssueCreator = require('..')

describe('create-an-issue', () => {
  let issueCreator, github

  beforeEach(() => {
    issueCreator = new IssueCreator()
    github = { issues: { create: jest.fn() } }

    issueCreator.tools.workspace = path.join(__dirname, 'fixtures')
    issueCreator.tools.context.payload = { repository: { owner: { login: 'JasonEtco' }, name: 'waddup' } }
    issueCreator.tools.createOctokit = jest.fn(() => github)
  })

  it('creates a new issue', async () => {
    await issueCreator.go()
    expect(github.issues.create).toHaveBeenCalled()
    expect(github.issues.create.mock.calls).toMatchSnapshot()
  })

  it('creates a new issue from a different template', async () => {
    issueCreator = new IssueCreator('.github/different-template.md')
    issueCreator.tools.workspace = path.join(__dirname, 'fixtures')
    issueCreator.tools.context.payload = { repository: { owner: { login: 'JasonEtco' }, name: 'waddup' } }
    issueCreator.tools.createOctokit = jest.fn(() => github)

    await issueCreator.go()
    expect(github.issues.create).toHaveBeenCalled()
    expect(github.issues.create.mock.calls[0][0].title).toBe('Different file')
  })

  it('creates a new issue with some template variables', async () => {
    issueCreator.template = '.github/variables.md'
    await issueCreator.go()
    expect(github.issues.create).toHaveBeenCalled()
    expect(github.issues.create.mock.calls).toMatchSnapshot()
  })

  it('creates a new issue with assignees, labels and a milestone', async () => {
    issueCreator.template = '.github/kitchen-sink.md'
    await issueCreator.go()
    expect(github.issues.create).toHaveBeenCalled()
    expect(github.issues.create.mock.calls).toMatchSnapshot()
  })
})
