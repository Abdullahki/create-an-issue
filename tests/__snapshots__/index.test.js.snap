// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`create-an-issue creates a new issue 1`] = `
Array [
  Array [
    Object {
      "assignees": Array [],
      "body": "Goodbye!",
      "labels": Array [],
      "milestone": undefined,
      "owner": "JasonEtco",
      "repo": "waddup",
      "title": "Hello!",
    },
  ],
]
`;

exports[`create-an-issue creates a new issue with some template variables 1`] = `
Array [
  Array [
    Object {
      "assignees": Array [],
      "body": "The repo waddup is the best repo.",
      "labels": Array [],
      "milestone": undefined,
      "owner": "JasonEtco",
      "repo": "waddup",
      "title": "Hello JasonEtco",
    },
  ],
]
`;
